src/waveform/renderers/waveformmark.cpp

```diff
@@ -565,3 +578,78 @@ QImage WaveformMark::generateImage(float devicePixelRatio) {
 
     return image;
 }
+
+QImage WaveformMark::generateEndImage(float devicePixelRatio) {
+    assert(needsEndImageUpdate());
+
+    const bool useIcon = m_endIconPath != "";
+
+    // Determine drawing geometries
+    const MarkerGeometry markerGeometry{"", useIcon, m_align, m_breadth, m_level};
+
+    m_label.setAreaRect(markerGeometry.labelRect());
+
+    // Create the image
+    QImage image{markerGeometry.getImageSize(devicePixelRatio),
+            QImage::Format_ARGB32_Premultiplied};
+    image.setDevicePixelRatio(devicePixelRatio);
+
+    // Fill with transparent pixels
+    image.fill(Qt::transparent);
+
+    QPainter painter;
+
+    painter.begin(&image);
+    painter.setRenderHint(QPainter::TextAntialiasing);
+    painter.setRenderHint(QPainter::Antialiasing);
+
+    painter.setWorldMatrixEnabled(false);
+
+    // Draw marker lines
+    const auto hcenter = markerGeometry.imageSize().width() / 2.f;
+    m_linePosition = static_cast<float>(hcenter);
```

Comment:
This looks weird. Isn't hcenter alread float. It is better to not use auto in thes cases.